org.jacoco.core/src/org/jacoco/core/internal/BytecodeVersion.java

```diff
@@ -56,25 +56,25 @@ public static void set(final byte[] b, final int version) {
 	}
 
 	/**
-	 * Returns given bytes of class if its major bytecode version is less that
-	 * {@link #V10}, otherwise returns copy where major version set to
-	 * {@link Opcodes#V9}.
+	 * Returns given bytes of class if its major bytecode version
+	 * {@link #MAX_VERSION} or less, otherwise returns copy where major version
+	 * set to {@link #MAX_VERSION}.
 	 * 
 	 * @param version
 	 *            version of bytecode
 	 * @param source
 	 *            bytes of class
-	 * @return given bytes of class if version is less than {@link #V10},
-	 *         otherwise copy where version set to {@link Opcodes#V9}
+	 * @return class definition with a version of at most {@link #MAX_VERSION}
 	 */
 	public static byte[] downgradeIfNeeded(final int version,
 			final byte[] source) {
-		if (V10 != version) {
+		// consider major version only (due to 1.1 anomaly)
+		if ((version & 0xff) <= MAX_VERSION) {
 			return source;
 		}
 		final byte[] b = new byte[source.length];
 		System.arraycopy(source, 0, b, 0, source.length);
-		set(b, Opcodes.V9);
+		set(b, MAX_VERSION);
```

Comment:
This trick not needed anymore given that ASM now supports and will continue supporting release+1.